src/common/string/filter/sequence-matching.ts

```diff
@@ -63,13 +76,46 @@ type FuzzyFilterSort = <T extends ScorableTextItem>(
 export const fuzzyFilterSort: FuzzyFilterSort = (filter, items) => {
   return items
     .map((item) => {
-      item.score = item.altText
-        ? fuzzySequentialMatch(filter, item.filterText, item.altText)
-        : fuzzySequentialMatch(filter, item.filterText);
+      const match = fuzzySequentialMatch(filter, item);
+
+      item.score = match?.score;
+      item.decoratedWords = match?.decoratedWords;
+
       return item;
     })
     .filter((item) => item.score !== undefined)
     .sort(({ score: scoreA = 0 }, { score: scoreB = 0 }) =>
       scoreA > scoreB ? -1 : scoreA < scoreB ? 1 : 0
     );
 };
+
+type MatchDecorator = (word: string, scores?: FuzzyScore) => TemplateResult[];
+export const decorateMatch: MatchDecorator = (word, scores) => {
+  if (!scores) {
+    return [html`${word}`];
+  }
+
+  const decoratedText: TemplateResult[] = [];
+  const matches = createMatches(scores);
+  let pos = 0;
+
+  let actualWord = "";
+  for (const match of matches) {
+    actualWord += word.substring(pos, match.start);
+    actualWord += `<span class="highlight-letter">${word.substring(
+      match.start,
+      match.end
+    )}</span>`;
+    pos = match.end;
+  }
+  actualWord += word.substring(pos);
+
+  const fragments = actualWord.split("::");
```

Comment:
Where are the `::` coming from?

Reply:
I'm adding them to the command items when generating them (in quick-bar.ts).

It's a bit tricky explaining why I'm using them, but here goes:

Let's say we type in "relfoo", hoping to find "Reload Foo Bar". The fuzzyMatcher finds the match, and generates this decorated text: `<b>Rel</b>oad <b>Foo</b> Bar`

Unfortunately, we can't render this as-is for command items cuz we're now rendering "Reload" and "Foo Bar" separately for command labels.

So we need some way of knowing which piece of `<b>Rel</b>oad <b>Foo</b> Bar` is the command label, and which part is the text.

I explored a few options, and the simplest one was to inject a delimiter between the category label and text before sending to the matcher. We then use that delimiter later to restore the separation of strings while preserving their decoration.

I decided against splitting on space cuz we may someday have a label with a space in it (e.g. "Server Control" or "Custom Integration X").